### PR TITLE
Feature/sprint 03 Show preloader

### PR DESCRIPTION
**Pull request by tarasbashuk**

### CODE REVIEW

File: client/src/pages/EmailNeedsConfirmation/EmailNeedsConfirmationPage.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  const classes = useStyles()
+  const history = useHistory()
+
+  const goBack = e => {
+    e.preventDefault()
+    history.goBack()
+  }
+
+  if (emailIsConfirmed) {
+    return <Redirect to='/' />
```
Comment (RostyslavStepanchuk) [inline client/src/pages/EmailNeedsConfirmation/EmailNeedsConfirmationPage.jsx]: Let's switch this also to history.goBack() cause we are not sure that visitor tried to enter root page when they were redirected to this one

File: client/src/actions/auth.js [hunk 2]
```diff
@@ -136,3 +144,24 @@ export const resetPassword = (email) => dispatch => {
...
+    type: START_LOADING
+  })
+
+  axios.get('/api/v1/users/email/confirm/' + token)
+    .then(res => {
+      if (res.status === 200) {
+        dispatch({
+          type: EMAIL_CONFIRMED
+        })
+        dispatch(loadUser())
```
Comment (RostyslavStepanchuk) [inline client/src/actions/auth.js]: We don't need to load user on email confirmed page, just confirm email

File: client/src/pages/EmailNeedsConfirmation/EmailNeedsConfirmationPage.jsx (hunk 1 above)
Comment (tarasbashuk) [reply client/src/pages/EmailNeedsConfirmation/EmailNeedsConfirmationPage.jsx]: Agree

File: client/src/actions/auth.js (hunk 2 above)
Comment (tarasbashuk) [reply client/src/actions/auth.js]: If a user closes our App after registration and then confirms an email there won't any info about the current user in the Store.

Comment (tarasbashuk) [reply client/src/actions/auth.js]: Or a user just goes somewhere else from the Confirmation page.

Comment (RostyslavStepanchuk) [reply client/src/actions/auth.js]: Wherever user goes, first action will be receive current user. As far emailConfirmed is a part of responseDTO you will always know if email was confirmed and will be able to save it to store.
Though, on this page we don't use any information about current user thus there is no need to get it

Comment (tarasbashuk) [reply client/src/actions/auth.js]: OK

**Review by RostyslavStepanchuk — APPROVED** [review]

**Pull request merged**